.travis.yml

```diff
@@ -94,6 +94,42 @@ matrix:
         # Report success since we have overridden default behaviour
         - bash -c "$STATUS" success "Local $NAME testing has passed"
 
+    - env:
+        - NAME=astyle
+      install:
+      - wget https://downloads.sourceforge.net/project/astyle/astyle/astyle%203.1/astyle_3.1_linux.tar.gz;
+        mkdir -p BUILD && tar xf astyle_3.1_linux.tar.gz -C BUILD;
+        pushd BUILD/astyle/build/gcc;
+        make;
+        export PATH=$PWD/bin:$PATH;
+        popd;
+      - astyle --version
+      script:
+        # only changed files to be checked for astyle
+        - git diff --name-only $TRAVIS_BRANCH | grep '.*\.\(h\|c\|hpp\|cpp\)' | fgrep -v -f .astyleignore | xargs -n 100 -I {} bash -c "astyle -n --options=.astylerc \"{}\"" > astyle.out;
+          if [ $(cat astyle.out | grep Formatted | wc -l) -ne 0 ]; then
+            git --no-pager diff;
+            echo "Please fix style issues as shown above";
+          else
+            echo "Coding style check OK";
+          fi
+      after_success:
+        # update status if we succeeded, compare with master if possible
+        - |
+          CURR=$(cat astyle.out | grep Formatted | wc -l)
+          PREV=$(curl https://api.github.com/repos/$TRAVIS_REPO_SLUG/status/master \
+              | jq -re "select(.sha != \"$TRAVIS_COMMIT\")
+                  | .statuses[] | select(.context == \"travis-ci/$NAME\").description
+                  | capture(\", (?<warnings>[0-9]+) warnings\").warnings" \
+              || echo 0)
+
+          STATUSM="Passed, ${CURR} warnings"
+          if [ "$PREV" -ne 0 ]
+          then
+              STATUSM="$STATUSM ($(python -c "print '%+d' % ($CURR-$PREV)") warnings)"
+          fi
```

Comment:
As is, once master has an astyle status, PRs will say "Passed, 4 warnings (+4 warnings)". We should probably remove this script from here if we don't want to compare with master.

Reply:
Why would we not want to compare with master? I would think that having warning delta would be useful, and part of this PR's intent.